src/faction_camp.cpp

```diff
@@ -3512,62 +3508,113 @@ void basecamp::start_combat_mission( const mission_id &miss_id, float exertion_l
     }
 }
 
-// the structure of this function has driven (at least) two devs insane
-// recipe_deck returns a map of recipe ids to descriptions
-// it first checks whether the mission id starts with the correct direction prefix,
-// and then search for the mission id without direction prefix in the recipes
-// if there's a match, the player has selected a crafting mission
-
-void basecamp::start_crafting( const std::string &type, const mission_id &miss_id )
+void basecamp::start_crafting( const mission_id &miss_id )
 {
-    const std::map<recipe_id, translation> &recipes = recipe_deck( type );
-    const auto it = recipes.find( recipe_id( miss_id.parameters ) );
-    if( it != recipes.end() ) {
-        const recipe &making = it->first.obj();
-
-        if( !making.deduped_requirements().can_make_with_inventory(
-                _inv, making.get_component_filter() ) ) {
-            popup( _( "You don't have the materials to craft that" ) );
-            return;
+    int num_to_make = 1;
+    npc dummy;
+    for( const recipe_id &some_known_recipe : recipe_deck_all() ) {
+        dummy.learn_recipe( &*some_known_recipe );
+    }
+    for( const npc_ptr &guy : assigned_npcs ) {
+        if( guy.get() ) {
+            // give the dummy a combination of all their skills
+            for( auto &skill_level_pair : guy->get_all_skills() ) {
+                if( skill_level_pair.second.level() > dummy.get_skill_level( skill_level_pair.first ) ) {
+                    dummy.set_skill_level( skill_level_pair.first, skill_level_pair.second.level() );
+                }
+            }
+            // and their proficiencies
+            for( auto &some_proficiency : guy->known_proficiencies() ) {
+                dummy.add_proficiency( some_proficiency, true, true );
+            }
         }
+    }
 
-        int batch_size = 1;
-        string_input_popup popup_input;
-        int batch_max = recipe_batch_max( making );
-        const std::string title = string_format( _( "Batch crafting %s [MAX: %d]: " ),
-                                  making.result_name( /*decorated=*/true ), batch_max );
-        popup_input.title( title ).edit( batch_size );
+    // Force dummy to always be able to see for crafting. We don't care if it's dark out.
+    dummy.add_effect( effect_HACK_camp_vision_for_npc, 1_turns, true );
 
-        if( popup_input.canceled() || batch_size <= 0 ) {
-            return;
-        }
-        if( batch_size > recipe_batch_max( making ) ) {
-            popup( _( "Your batch is too large!" ) );
-            return;
+    validate_sort_points();
+    form_crafting_inventory( get_camp_map() );
+
```

Comment:
```suggestion
```

Reply:
I'm having trouble seeing what this suggestion is, github's diff highlighting is not making it clear. You're suggesting I remove the sort point validation and formation of the crafting inventory before we call select_crafter_and_crafting_recipe?

Reply:
That's VERY weird and obviously not what I had in mind. GitHub sometimes confuses me. Let's close this suggestion.